components/elm/src/data_types/ColumnDataType.F90

```diff
@@ -5763,6 +5773,34 @@ subroutine col_cf_init(this, begc, endc, carbon_type)
            call hist_addfld1d (fname='HR', units='gC/m^2/s', &
                 avgflag='A', long_name='total heterotrophic respiration', &
                  ptr_col=this%hr)
+! Shijie
+          this%hrv_deadstemc_to_prod10c(begc:endc) = spval
+          call hist_addfld1d (fname='HRV_DEADSTEMC_TO_PROD10C', units='gC/m^2/s', &
+               avgflag='A', long_name='flux into 10-yr wood product C', &
+               ptr_col=this%hrv_deadstemc_to_prod10c, default='inactive')
+
+          this%hrv_deadstemc_to_prod100c(begc:endc) = spval
+          call hist_addfld1d (fname='HRV_DEADSTEMC_TO_PROD100C', units='gC/m^2/s', &
+               avgflag='A', long_name='flux into 100-yr wood product C', &
+               ptr_col=this%hrv_deadstemc_to_prod100c, default='inactive')
+
+          this%nep(begc:endc) = spval
+          call hist_addfld1d (fname='NEP', units='gC/m^2/s', &
+               avgflag='A', long_name='net ecosystem production, excludes fire, landuse, and harvest flux, positive for sink', &
+                ptr_col=this%nep)
+ 
+          this%nbp(begc:endc) = spval
+          call hist_addfld1d (fname='NBP', units='gC/m^2/s', &
+               avgflag='A', long_name='net biome production, includes fire, landuse, and harvest flux, positive for sink', &
+                ptr_col=this%nbp)
+ 
+          this%nee(begc:endc) = spval
+          call hist_addfld1d (fname='NEE', units='gC/m^2/s', &
```

Comment:
will NEE also be defined further down? see https://github.com/E3SM-Project/E3SM/blob/e96fc25b2d223d21530fd78ab5212feac1ecf44c/components/elm/src/data_types/ColumnDataType.F90#L6121

Reply:
same with NEP and NBP, if already defined somewhere else that is active, lets use the same definition

Reply:
This repetitive definition is actually in a different IF-ELSE blocks (.not. FATES .and. C12). I'm not sure how to merge them?

Reply:
ok, I thought that was an end if, I see its an else if now.  I have no problems with the current formulation then.  An alternative would be to move all instances of NEP, NEE and NBP to either the very beginning of the routine, or the very end, which would be outside the if fates (and wrap with ane if c12).